onthegomap/planetiler
# Custom PBF parsing

Previously planetiler depended on graphhopper utilities for PBF parsing. This change pulls in PBF parsing so we can have more control over the threading model, performance optimizations, and eventually support for locations on ways.

Left to do:

- [x] move header parsing into PBF parser/decoder
- [x] tests for PBF parser/decoder
- [x] comments/cleanup
- [x] check for ascending element IDs, nodes before ways, ways before rels
- [x] performance/low memory test

Then after this change:

- remove graphhopper dependency and update HPPC to the latest version
- try to parallelize element handling in pass1
- mmap improvements: use bigger mmap segments, madvise random
